lib/hashie/extensions/coercion.rb

```diff
@@ -3,27 +3,70 @@ class CoercionError < StandardError; end
 
   module Extensions
     module Coercion
-      CORE_TYPES = {
-        Integer    => :to_i,
-        Float      => :to_f,
-        Complex    => :to_c,
-        Rational   => :to_r,
-        String     => :to_s,
-        Symbol     => :to_sym
-      }
-
       ABSTRACT_CORE_TYPES = {
         Integer => [Fixnum, Bignum],
         Numeric => [Fixnum, Bignum, Float, Complex, Rational]
       }
 
-      def self.included(base)
-        base.send :include, InstanceMethods
-        base.extend ClassMethods # NOTE: we wanna make sure we first define set_value_with_coercion before extending
+      class CoercionSystemIncludeBuilder < Module
```

Comment:
I'd like this extracted to its own file at `lib/hashie/extensions/coercion/coercion_system_include_builder.rb`.

Also, why is this a class that inherits from `Module`? Shouldn't it really be one or the other?